frontend/src/components/commons/QuitTest.jsx

```diff
@@ -0,0 +1,168 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { Button } from "react-bootstrap";
+import PopupBox, { BUTTON_TYPE, BUTTON_STATE } from "../commons/PopupBox";
+import SystemMessage, { MESSAGE_TYPE } from "../commons/SystemMessage";
+import LOCALIZE from "../../text_resources";
+import { deactivateTest } from "../../modules/TestStatusRedux";
+
+const styles = {
+  button: {
+    marginRight: 15
+  },
+  hr: {
+    width: "100%",
+    borderTop: "2px solid #96a8b2",
+    margin: "16px 0 16px 0"
+  },
+  checkboxZone: {
+    paddingTop: 8
+  },
+  startTestBtn: {
+    textAlign: "center",
+    marginTop: 32
+  }
+};
+
+const quitConditions = () => {
+  return [
+    { text: LOCALIZE.emibTest.testFooter.quitTestPopupBox.checkboxOne, checked: false },
+    { text: LOCALIZE.emibTest.testFooter.quitTestPopupBox.checkboxTwo, checked: false },
+    { text: LOCALIZE.emibTest.testFooter.quitTestPopupBox.checkboxThree, checked: false }
+  ];
+};
+
+class QuitTest extends Component {
+  static propTypes = {
+    // Provided by Redux
+    deactivateTest: PropTypes.func.isRequired,
+    isTestActive: PropTypes.bool
+  };
+
+  state = {
+    showQuitPopup: false,
+    quitConditions: quitConditions()
+  };
+
+  closePopup = () => {
+    this.setState({ showQuitPopup: false });
+    //reset all checkbox states on close
+    this.resetCheckboxStates();
+  };
+
+  openQuitPopup = () => {
+    this.setState({ showQuitPopup: true });
+  };
+
+  resetCheckboxStates = () => {
+    this.setState({ quitConditions: quitConditions() });
```

Comment:
hmm...  though this looks like a valid solution for the above problem.... interesting....

Reply:
ok, the issue only occurs IF YOU CLICK THE BACKGROUND to close the dialog. If you hit the button it works as expected....

Reply:
replicated on master... resolving